fili-security/src/main/java/com/yahoo/bard/webservice/web/security/RoleDimensionApiFilterRequestMapper.java

```diff
@@ -0,0 +1,161 @@
+ * based on roles for that user.
+ */
+public class RoleDimensionApiFilterRequestMapper extends ChainingRequestMapper<DataApiRequest> {
+
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractBinderFactory.class);
+
+    public static final String SECURITY_SIGNUP_MESSAGE_KEY = "security_signup_message";
+    public static final String DEFAULT_SECURITY_MESSAGE = "Your security settings do not permit access to this " +
+            "resource.  Please contact your user administration for access.";
+
+    private final static String UNAUTHORIZED_USER_MESSAGE = SYSTEM_CONFIG.getStringProperty(
+            SYSTEM_CONFIG.getPackageVariableName(SECURITY_SIGNUP_MESSAGE_KEY),
+            DEFAULT_SECURITY_MESSAGE
+    );
+
+    private final String unauthorizedHttpMessage;
+
+    Dimension dimension;
+    Map<String, Set<ApiFilter>> roleApiFilters;
+
+    /**
+     * Constructor.
+     *
+     * @param resourceDictionaries  The dictionaries to use for request mapping.
+     * @param dimension  The dimension whose roles are being matched
+     * @param roleApiFilters  ApiFilters by role for a given dimension
+     */
+    public RoleDimensionApiFilterRequestMapper(
+            ResourceDictionaries resourceDictionaries,
+            Dimension dimension,
+            Map<String, Set<ApiFilter>> roleApiFilters
+    ) {
+        this(resourceDictionaries, dimension, roleApiFilters, null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param resourceDictionaries  The dictionaries to use for request mapping.
+     * @param dimension  The dimension whose roles are being matched
+     * @param roleApiFilters  ApiFilters by role for a given dimension
+     * @param next  The next request mapper to process this ApiRequest
+     */
+    public RoleDimensionApiFilterRequestMapper(
+            final ResourceDictionaries resourceDictionaries,
+            Dimension dimension,
+            Map<String, Set<ApiFilter>> roleApiFilters,
+            ChainingRequestMapper<DataApiRequest> next
+    ) {
+        super(resourceDictionaries, next);
+        this.dimension = dimension;
+        this.roleApiFilters = roleApiFilters;
+        unauthorizedHttpMessage = DIMENSION_MISSING_MANDATORY_ROLE.format(dimension.getApiName());
+    }
+
+    @Override
+    public DataApiRequest internalApply(DataApiRequest request, ContainerRequestContext context)
+            throws RequestValidationException {
+        SecurityContext securityContext = context.getSecurityContext();
+
+        Set<ApiFilter> mergedSecurityFilters = unionMergeFilterValues(
+                roleApiFilters.keySet().stream()
+                        .filter(securityContext::isUserInRole)
+                        .map(roleApiFilters::get)
+                        .flatMap(Set::stream)
+        );
+
+        if (mergedSecurityFilters.size() == 0) {
```

Comment:
I don't think we should rely on the `mergedSecurityFilters` being zero to signal that the user does not have a valid role. What if a customer wants to have some sort of god role that can see everything? Then they'd probably much rather not attach any filters at all, then be forced to attach a filter that contains _everything_.

While it's less elegant, we're probably better off first finding out if the user has at least one of the specified roles and then merging the api filters only if they have one of those roles.